components/apimgt/org.wso2.carbon.apimgt.usage/org.wso2.carbon.apimgt.usage.client/src/main/java/org/wso2/carbon/apimgt/usage/client/UsageClient.java

```diff
@@ -0,0 +1,217 @@
+* KIND, either express or implied. See the License for the
+* specific language governing permissions and limitations
+* under the License.
+*
+*/
+package org.wso2.carbon.apimgt.usage.client;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.apimgt.api.APIManagementException;
+import org.wso2.carbon.apimgt.api.APIProvider;
+import org.wso2.carbon.apimgt.api.model.API;
+import org.wso2.carbon.apimgt.api.model.APIStatus;
+import org.wso2.carbon.apimgt.impl.APIManagerAnalyticsConfiguration;
+import org.wso2.carbon.apimgt.impl.APIManagerConfiguration;
+import org.wso2.carbon.apimgt.impl.APIManagerFactory;
+import org.wso2.carbon.apimgt.impl.internal.ServiceReferenceHolder;
+import org.wso2.carbon.apimgt.impl.utils.APIUtil;
+import org.wso2.carbon.apimgt.usage.client.exception.APIMgtUsageQueryServiceClientException;
+import org.wso2.carbon.apimgt.usage.client.pojo.APIFirstAccess;
+import org.wso2.carbon.apimgt.usage.client.pojo.SubscriberCountByAPIs;
+import org.wso2.carbon.context.PrivilegedCarbonContext;
+import org.wso2.carbon.utils.multitenancy.MultitenantConstants;
+import org.wso2.carbon.utils.multitenancy.MultitenantUtils;
+
+import java.lang.reflect.InvocationTargetException;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * usageClient class it use to expose the Statistic class instance. it responsible to make instance of the class that is provided by the api-manager.xml
+ */
+public class UsageClient {
+    private static final Log log = LogFactory.getLog(UsageClient.class);
+    private static APIUsageStatisticsClient usageStatisticsClient;
+
+    /**
+     * central point to initialise datasources or related configuration done by the admin-dashboard analytics section
+     *
+     * @throws APIMgtUsageQueryServiceClientException
+     */
+    public static void initializeDataSource() throws APIMgtUsageQueryServiceClientException {
+
+        try {
+            APIUsageStatisticsClient client = UsageClient.getStatisticClient("");
+            client.initializeDataSource();
+        } catch (APIMgtUsageQueryServiceClientException e) {
+            throw new APIMgtUsageQueryServiceClientException("Error in initializing data sources", e);
+        }
+
+    }
+
+    /**
+     * central public method used to get the instance if the statistic client
+     *
+     * @return return the APIUsageStatisticsClient implementation
+     * @throws APIMgtUsageQueryServiceClientException if error in creating instance
+     */
+    public static APIUsageStatisticsClient getClient(String user) throws APIMgtUsageQueryServiceClientException {
+        if (isDataPublishingEnabled()) {
+            try {
+                return UsageClient.getStatisticClient(user);
+            } catch (APIMgtUsageQueryServiceClientException e) {
+                throw new APIMgtUsageQueryServiceClientException("Error getting Statistics usage client instance", e);
```

Comment:
fixed in commit: https://github.com/wso2/carbon-apimgt/commit/c5071cbc7f3c39fed2489c288179e0fe93507544